openstack/messaging/v2/queues/results.go

```diff
@@ -0,0 +1,67 @@
+package queues
+
+import (
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/pagination"
+)
+
+// QueuePage contains a single page of all queues from a List operation.
+type QueuePage struct {
+	pagination.LinkedPageBase
+}
+
+// CreateResult is the response of a Create operation.
+type CreateResult struct {
+	gophercloud.ErrResult
+}
+
+// Queue represents a messaging queue.
+type Queue struct {
+	Href          string       `json:"href"`
+	Methods       []string     `json:"methods"`
+	Name          string       `json:"name"`
+	Paths         []string     `json:"paths"`
+	ResourceTypes []string     `json:"resource_types"`
+	Metadata      QueueDetails `json:"metadata"`
+}
+
+// QueueDetails represents the metadata of a queue.
+type QueueDetails struct {
+	DeadLetterQueue           string `json:"_dead_letter_queue"`
+	DeadLetterQueueMessageTTL int    `json:"_dead_letter_queue_messages_ttl"`
+	DefaultMessageDelay       int    `json:"_default_message_delay"`
+	DefaultMessageTTL         int    `json:"_default_message_ttl"`
+	MaxClaimCount             int    `json:"_max_claim_count"`
+	MaxMessagesPostSize       int    `json:"_max_messages_post_size"`
+	Description               string `json:"description"`
+	Flavor                    string `json:"flavor"`
+}
```

Comment:
Since you can create a Queue with arbitrary key/value pairs, those extra metadata pairs will be returned here, too. So we need to extract all of them into a "bucket" field. There's an internal function called [`RemainingKeys`](https://github.com/gophercloud/gophercloud/blob/master/openstack/identity/v3/users/results.go#L72) that can handle this. Let me know if you need help implementing it.

With that in mind, I think we should remove `Description`. I think Flavor is used in other areas of Zaqar, so it should be fine to keep as a "standard" field.

Reply:
Done, I also expanded the unit tests to test to test this.